parser/parser.go

```diff
@@ -64,12 +64,67 @@ type Node interface {
 }
 
 type AST struct {
-	Select  *Select `(   "SELECT"  @@  `
-	Insert  *Insert `  | "INSERT"  @@  `
-	Replace *Insert `  | "REPLACE" @@ )`
-	End     string  `( ";" )?`
+	Select      *Select      `(   "SELECT"         @@`
+	Insert      *Insert      `  | "INSERT"         @@`
+	Replace     *Insert      `  | "REPLACE"        @@`
+	CreateTable *CreateTable `  | "CREATE" "TABLE" @@ ) ";"?`
 }
 
+type CreateTable struct {
+	Table   string              `@(Ident | QuotedIdent) "("`
+	Entries []*CreateTableEntry `@@ ("," @@)* ")"`
+}
+
+func (c *CreateTable) node() {}
+
+type CreateTableEntry struct {
+	GlobalSecondaryIndex  *GlobalSecondaryIndex  `  @@`
+	LocalSecondaryIndex   *LocalSecondaryIndex   `| @@`
+	ProvisionedThroughput *ProvisionedThroughput `| @@`
+	Attr                  *TableAttr             `| @@` // Must be last.
+}
+
+func (c *CreateTableEntry) node() {}
+
+type ProvisionedThroughput struct {
+	ReadCapacityUnits  int64 `"PROVISIONED" "THROUGHPUT" "READ" @Number`
+	WriteCapacityUnits int64 `"WRITE" @Number`
+}
+
+func (p *ProvisionedThroughput) node() {}
+
+type GlobalSecondaryIndex struct {
+	Name                  string                 `"GLOBAL" "SECONDARY" "INDEX" @(Ident | QuotedIdent)`
+	PartitionKey          string                 `"HASH" "(" @(Ident | QuotedIdent) ")"`
+	SortKey               string                 `"RANGE" "(" @(Ident | QuotedIdent) ")"`
+	Projection            *Projection            `"PROJECTION" @@`
```

Comment:
I wonder if we should make this optional and default to `KEYS_ONLY`?

Reply:
I also wondered that, but it didn't seem like there was a clear default choice. KEYS_ONLY may not provide enough information while `ALL` might not be performant. I think leave it for now and we can provide a default now without breaking backwards compat.